src/Orleans/Serialization/OrleansJsonSerializer.cs

```diff
@@ -237,5 +233,29 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
         }
     }
 
+    class GuidConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType.IsAssignableFrom(typeof(Guid));
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            Guid guid = (Guid)value;
+            writer.WriteStartObject();
+            writer.WritePropertyName("Guid");
+            writer.WriteValue(value.ToString());
+            writer.WriteEndObject();
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            JObject jo = JObject.Load(reader);
+            Guid guid = Guid.Parse(jo["Guid"].ToObject<string>());
+            return guid;
+        }
```

Comment:
but what converter is it on? The key part is what in function CanConvert. It is applied per type.